chain/x/zoracle/exec_env.go

```diff
@@ -63,20 +116,20 @@ func (env *ExecutionEnvironment) GetValidatorAddress(validatorIndex int64) ([]by
 	return env.request.RequestedValidators[validatorIndex], nil
 }
 
-func (env *ExecutionEnvironment) GetMaximumResultSize() int64 {
-	return int64(env.keeper.GetParam(env.ctx, KeyMaxResultSize))
-}
-
-func (env *ExecutionEnvironment) GetMaximumCalldataOfDataSourceSize() int64 {
-	return int64(env.keeper.GetParam(env.ctx, KeyMaxCalldataSize))
-}
-
 func (env *ExecutionEnvironment) RequestExternalData(
 	dataSourceID int64,
 	externalDataID int64,
 	calldata []byte,
-) error {
-	return env.keeper.AddNewRawDataRequest(env.ctx, env.requestID, types.ExternalID(externalDataID), types.DataSourceID(dataSourceID), calldata)
+) int64 {
```

Comment:
This should return error to be consistent with other functions (see GetExternalData as example). In `resolver` you return -1 if error is not nil.